CI failures: Cannot run program "./cs": error=2, No such file or directory

We have a few spurious failures of CI involving `CoursierScalaTests`. Usually rerunning the tests fixes the issue.

* https://github.com/lampepfl/dotty/actions/runs/4427028539/jobs/7768013294
* https://github.com/lampepfl/dotty/actions/runs/4430193399/jobs/7771618203
* https://github.com/lampepfl/dotty/actions/runs/4414553285/jobs/7736411129
* ...


```
[info] Test dotty.tools.coursier.CoursierScalaTests.allTests started
Error:  Test dotty.tools.coursier.CoursierScalaTests.allTests failed: java.io.IOException: Cannot run program "./cs": error=2, No such file or directory, took 0.088 sec
Error:      at java.lang.ProcessBuilder.start(ProcessBuilder.java:1142)
Error:      at java.lang.ProcessBuilder.start(ProcessBuilder.java:1073)
Error:      at scala.sys.process.ProcessBuilderImpl$Simple.run(ProcessBuilderImpl.scala:85)
Error:      at scala.sys.process.ProcessBuilderImpl$AbstractBuilder.run(ProcessBuilderImpl.scala:119)
Error:      at scala.sys.process.ProcessBuilderImpl$AbstractBuilder.$anonfun$runBuffered$1(ProcessBuilderImpl.scala:205)
Error:      at scala.runtime.java8.JFunction0$mcI$sp.apply(JFunction0$mcI$sp.scala:17)
Error:      at scala.sys.process.ProcessLogger$$anon$1.buffer(ProcessLogger.scala:103)
Error:      at scala.sys.process.ProcessBuilderImpl$AbstractBuilder.runBuffered(ProcessBuilderImpl.scala:205)
Error:      at scala.sys.process.ProcessBuilderImpl$AbstractBuilder.!(ProcessBuilderImpl.scala:146)
Error:      at dotty.tools.coursier.CoursierScalaTests$.execCmd(CoursierScalaTests.scala:154)
Error:      at dotty.tools.coursier.CoursierScalaTests$.csCmd(CoursierScalaTests.scala:169)
Error:      at dotty.tools.coursier.CoursierScalaTests$.csScalaCmd(CoursierScalaTests.scala:158)
Error:      at dotty.tools.coursier.CoursierScalaTests.scriptArgs$1(CoursierScalaTests.scala:42)
Error:      at dotty.tools.coursier.CoursierScalaTests.allTests(CoursierScalaTests.scala:55)
Error:      at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
Error:      at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
Error:      at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Error:      at java.lang.reflect.Method.invoke(Method.java:567)
Error:      ...
Error:  Caused by: java.io.IOException: error=2, No such file or directory
Error:      at java.lang.ProcessImpl.forkAndExec(Native Method)
Error:      at java.lang.ProcessImpl.this(ProcessImpl.java:313)
Error:      at java.lang.ProcessImpl.start(ProcessImpl.java:244)
Error:      at java.lang.ProcessBuilder.start(ProcessBuilder.java:1109)
Error:      ... 51 more
```

Now we got a different error 

```
Error:  Test dotty.tools.coursier.CoursierScalaTests failed: java.lang.AssertionError: Failed to run curl -fLo cs https://git.io/coursier-cli-linux, exit code: 22, output:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
Error:                                   Dload  Upload   Total   Spent    Left  Speed
Error:  
Error:    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
Error:    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
Error:  
Error:    0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
Error:  
Error:    0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
Error:  
Error:    0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
Error:  curl: (22) The requested URL returned error: 500 , took 0.0 sec
Error:      at dotty.tools.coursier.CoursierScalaTests$.runAndCheckCmd$1(CoursierScalaTests.scala:178)
Error:      at dotty.tools.coursier.CoursierScalaTests$.setup(CoursierScalaTests.scala:180)
Error:      at dotty.tools.coursier.CoursierScalaTests.setup(CoursierScalaTests.scala)
Error:      at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
Error:      at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
Error:      at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Error:      at java.lang.reflect.Method.invoke(Method.java:567)
Error:      ...
```
https://github.com/lampepfl/dotty/actions/runs/4487258054/jobs/7890504165?pr=17059

--- a/compiler/test-coursier/dotty/tools/coursier/CoursierScalaTests.scala
+++ b/compiler/test-coursier/dotty/tools/coursier/CoursierScalaTests.scala
@@ -170,12 +170,16 @@ object CoursierScalaTests:
 
   /** Get coursier script */
   @BeforeClass def setup(): Unit =
-    val ver = execCmd("uname")._2.head.replace('L', 'l').replace('D', 'd')
+    val launcherLocation = "https://github.com/coursier/launchers/raw/master"
+    val launcherName = execCmd("uname")._2.head.toLowerCase match
+      case "linux" => "cs-x86_64-pc-linux"
+      case "darwin" => "cs-x86_64-apple-darwin"
+      case other => fail(s"Unsupported OS for coursier launcher: $other")
 
     def runAndCheckCmd(cmd: String, options: String*): Unit =
       val (code, out) = execCmd(cmd, options*)
       if code != 0 then
         fail(s"Failed to run $cmd ${options.mkString(" ")}, exit code: $code, output: ${out.mkString("\n")}")
 
-    runAndCheckCmd("curl", s"-fLo cs https://git.io/coursier-cli-$ver")
+    runAndCheckCmd("curl", s"-fLo cs $launcherLocation/$launcherName")
     runAndCheckCmd("chmod", "+x cs")

